src/app/product-detail/product-detail.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { ProductService } from '../api/products.service';
+import { IProduct } from '../Interface/products';
+
+@Component({
+  selector: 'app-product-detail',
+  templateUrl: './product-detail.component.html',
+  styleUrls: ['./product-detail.component.scss']
+})
+export class ProductDetailComponent implements OnInit, OnDestroy {
+
+  productId: number = 0;
+
+  products: IProduct[] = [];
+
+  sub!: Subscription;
+
+  isDataProvided: boolean = true;
```

Comment:
The isDataProvided is duplicated and imo not necessary.
If this.productService.productData$.subscribe does not return values it is no provided implicitly. 
I think the way to go would be to initialise the data somewhere globally.

Reply:
I've removed the variable 'isDataProvided'. You are right, it is not necessary. Now it looks like: this.productService.productData$.subscribe(data => if(data.length === 0){ ...} else { ... }. Is it a right way?